### DISCUSSION TITLE
Pythonへの移行について

### DISCUSSION THREAD

**Discussion by Tauzu** [Enehentaro/droplets_simulation#120] [Ideas]
# 飛沫計算、Pythonでリメイクしませんか。

## メリット
- 可読性・保守性の向上
- 近傍探索あたりのPythonライブラリが使える
- Fortranという負の遺産からの解放
- リスト、辞書、タプル等、便利な型が使える
- クラスも簡単に定義できる！

## デメリット
- Pythonについてかなり勉強しなければならない
- 計算が遅くなる恐れあり
  - Numpy, SciPyあたりを使えば高速化できそう
  - 所詮は飛沫計算なので、遅くなるとしてもたかが知れてる
- 恩恵を受けられるのはおそらく半年後くらいからか

**Comment by Tauzu** [comment]
## 自作する必要があると思われるライブラリ
探せばどっかのリポジトリに転がってるかも

- 飛沫に関する定義
  - 微分方程式（蒸発・運動）
  - 合体判定（優先度低め）
  - 参照セル探索
  - ボックスカウント
- 流れ場クラス
  - セル隣接関係解決（参照セル探索用）
  - 境界面の定義（付着判定用）

## 自作する必要のないライブラリ
- VTK
- etc.

**Comment by hirotori** [comment]
> セル隣接関係解決（参照セル探索用）

ここを以前`vtk`でやろうとしたけど, whileループとfor文回す関係でFortranよりはかなり時間かかる. 
格子の読み込みは大きな負荷ではなさそう.

3Dメッシュ関係のライブラリなら[Open3D](http://www.open3d.org/docs/release/index.html)もあるが, 触ってないので(無責任!!)わからん.

非構造メッシュが全ての元凶(異論は認める).

**Reply by Tauzu** [reply]
構造格子に戻る時が来たか・・・。
